src/py/flwr/server/app.py

```diff
@@ -258,18 +272,33 @@ def run_fleet_api() -> None:
 
     # Start Fleet API
     if args.fleet_api_type == "rest":
-        fleet_thread = threading.Thread(
-            target=_run_fleet_api_rest,
-            args=(args.rest_fleet_api_address, state_factory),
-        )
-        fleet_thread.start()
-        bckg_threads.append(fleet_thread)
+        parsed_address = parse_address(args.rest_fleet_api_address)
```

Comment:
Can we do the validation in between `event(...)` and state initialization?

Reply:
@danieljanes I changed it, but it means that we know check 2 times if we are using rest or grpc.

Reply:
@charlesbvll good point, I agree that the previous version was nicer